### PR TITLE
fix(ButtonMenu): doesn't support ghost kind

### PR DESCRIPTION
**Pull request by Ratheeshrajan**
Contributes to #1770 

ButtonMenu doesn't support ghost & tertiary kind

#### What did you change?
- Button kind class name was hardcoded as `${carbon.prefix}--btn--primary` and it's replaced with `${carbon.prefix}--btn--${size}` to accept kind props.
- Along with this, all the three kinds of button menus(primary, tertiary and ghost) are displayed in StoryBook for reference.
#### How did you test and verify your work?
Storybook.

### CODE REVIEW

File: packages/cloud-cognitive/src/components/ButtonMenu/ButtonMenu.js [hunk 1]
```diff
@@ -116,6 +118,11 @@ ButtonMenu.propTypes = {
    */
   iconDescription: Button.propTypes.iconDescription,
 
+  /**
+   * The three types the menu button supports: primary, tertiary and ghost.
+   */
+  kind: Button.propTypes.kind,
```
Comment (elycheea) [inline packages/cloud-cognitive/src/components/ButtonMenu/ButtonMenu.js:124]:
```suggestion
  kind: PropTypes.oneOf(['primary', 'tertiary', 'ghost']),
,
```

**Review by elycheea — CHANGES_REQUESTED** [review]
Rather than showing the three kinds of button menus in Storybook, I think it’s better to use the add them as options in Storybook controls.

image

Since we’ve hardcoded the `kind` in Storybook, the examples break when you try to modify `kind` in the controls tab. We may want to show just one button menu but allow changing of the prop. I think we have examples in SidePanel for setting the size or placement where we have known options.

File: packages/cloud-cognitive/src/components/ButtonMenu/ButtonMenu.js (hunk 1 above)
Comment (Ratheeshrajan) [reply packages/cloud-cognitive/src/components/ButtonMenu/ButtonMenu.js]: @elycheea Many thanks for reviewing this. Proposed changes were implemented.

**Review by elycheea — COMMENTED** [review]
Looks good but we’re still running into some lint errors in CI.
Should be fixed after a `yarn format` though.

**Comment by Ratheeshrajan** [comment]
@elycheea fixed the lint failure.

**Review by lee-chase — APPROVED** [review]

**Review by elycheea — APPROVED** [review]

**Pull request merged**